primap2/pm2io/_data_reading.py

```diff
@@ -1490,64 +1523,105 @@ def harmonize_units(
     if unit_col is None:
         unit_col = dim_aliases.get("unit", "unit")
 
+    # find basic entities for all entities and make a list
     entities = data[entity_col].unique()
+    basic_entities = {}
+
     # print(entities)
     for entity in entities:
         # check if GWP given in entity
+        # print(f"entity: {entity}")
         gwp_match = re.findall(r"\(([A-Z0-9]*)\)$", entity)
         if gwp_match:
             gwp_to_use = gwp_match[0]
-            basic_entity = re.findall(r"^[^()\s]*", entity)
+            basic_entity = re.findall(r"^[^\(\)\s]*", entity)
```

Comment:
Within `[]`, you don't have to escape braces, so this change is probably wrong, because the effective exclusion group here will be "any character which is not `(`, `)`, or `\`". I'm not sure if `\` in the entity is something we necessarily want to support, but also no reason to break it here, I think.

Reply:
Thanks for pointing that out. That was not a feature but just my lack of knowledge that the escaping is not necessary.